test/integration/smart-loan.test.ts

```diff
@@ -424,5 +430,170 @@ describe('Smart loan', () => {
+
+      usdTokenDecimalPlaces = await usdTokenContract.decimals();
+      linkTokenDecimalPlaces = await linkTokenContract.decimals();
+
+      MOCK_AVAX_PRICE = 100000;
+      MOCK_PRICES = [
+        {
+          symbol: 'USD',
+          value: MOCK_AVAX_PRICE * fromWei(await exchange.getEstimatedAVAXForERC20Token(toWei("1", usdTokenDecimalPlaces), usdTokenAddress))
+        },
+        {
+          symbol: 'LINK',
+          value: MOCK_AVAX_PRICE * fromWei(await exchange.getEstimatedAVAXForERC20Token(toWei("1", usdTokenDecimalPlaces), usdTokenAddress))
+        },
+        {
+          symbol: 'AVAX',
+          value: MOCK_AVAX_PRICE
+        }
+      ]
+
+      await pool.initialize(fixedRatesCalculator.address, borrowersRegistry.address, ZERO, ZERO);
+      await pool.connect(depositor).deposit({value: toWei("1000")});
+    });
+
+    it("should deploy a smart loan", async () => {
+      loan = await (new SmartLoan__factory(owner).deploy());
+      loan.initialize(supportedAssets.address, exchange.address, pool.address);
+
+      wrappedLoan = WrapperBuilder
+        .mockLite(loan)
+        .using(
+          () => {
+            return {
+              prices: MOCK_PRICES,
+              timestamp: Date.now()
+            }
+          })
+
+      await wrappedLoan.authorizeProvider();
+    });
+
+    it("should fund a loan", async () => {
+      await wrappedLoan.fund({value: toWei("100")});
+    });
+
+    it("should borrow funds", async () => {
+      await wrappedLoan.borrow(toWei("300"));
+
+      expect(fromWei(await wrappedLoan.getTotalValue())).to.be.equal(400);
+      expect(fromWei(await wrappedLoan.getDebt())).to.be.closeTo(300, 0.1);
+      expect(await wrappedLoan.getSolvencyRatio()).to.be.closeTo(BigNumber.from(1333), 1);
+    });
+
+    it("should invest", async () => {
+
+
+      await wrappedLoan.invest(toBytes32('USD'), toWei("9000", usdTokenDecimalPlaces));
+      await wrappedLoan.invest(toBytes32('LINK'), toWei("10000", usdTokenDecimalPlaces));
+
+      let balances = await wrappedLoan.getAllAssetsBalances();
+
+      const currentUSDTokenBalance = balances[0];
+      const currentLINKTokenBalance = balances[1];
+      expect(currentUSDTokenBalance).to.be.equal(toWei("9000", usdTokenDecimalPlaces));
+      expect(currentLINKTokenBalance).to.be.equal(toWei("10000", usdTokenDecimalPlaces));
+    });
+
+    it("should fail a sellout attempt", async () => {
+
+      let UPDATED_MOCK_PRICES = MOCK_PRICES.map(
+        (token: any) => {
+          if (token.symbol == 'USD' || token.symbol == 'LINK') {
+            token.value = 0.6 * token.value;
+          }
+          return token;
+        }
+      );
+
+      let loanUpdated = WrapperBuilder
+        .mockLite(loan)
+        .using(
+          () => {
+            return {
+              prices: UPDATED_MOCK_PRICES,
+              timestamp: Date.now()
+            }
+          })
+
+      let loan_solvent = await loanUpdated.isSolvent();
```

Comment:
Can we calculate solvency at this point? It would be more clear in the test that solvency is between 1.05 and 1.2